### PR TITLE
move chain to schema

### PR DESCRIPTION
**Pull request by hwchase17**

### CODE REVIEW

File: langchain/schema/chain.py [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import yaml
+from pydantic import Field, root_validator, validator
+
+import langchain
+from langchain.load.dump import dumpd
+from langchain.load.serializable import Serializable
+from langchain.schema import RUN_KEY, BaseMemory, RunInfo
```
Comment (baskaryan) [inline langchain/schema/chain.py:16]: will this cause issues?

Comment (hwchase17) [reply langchain/schema/chain.py:16]: yup should be direct import, good catch

Comment (baskaryan) [reply langchain/schema/chain.py]: lgtm otherwise 👍

File: langchain/schema/__init__.py
Comment (baskaryan) [inline langchain/schema/__init__.py:1]: actually one more comment, do we want to expose Chain here?

**Pull request closed without merging**